comp/dogstatsd/server/server.go

```diff
@@ -190,16 +194,32 @@ func initTelemetry(cfg config.Reader, logger logComponent.Component) {
 // TODO: (components) - remove once serverless is an FX app
 //
 //nolint:revive // TODO(AML) Fix revive linter
-func NewServerlessServer() Component {
-	return newServerCompat(config.Datadog, logComponentImpl.NewTemporaryLoggerWithoutInit(), replay.NewServerlessTrafficCapture(), serverdebugimpl.NewServerlessServerDebug(), true)
+func NewServerlessServer(demux aggregator.Demultiplexer) (ServerlessDogstatsd, error) {
```

Comment:
We can also move this constructor to it's own file (with the `Stop` method) for consistency.